### PR TITLE
Update theurgy.lic

### PR DESCRIPTION
**Pull request by drboldwin**
#882

as well as accounting for cyclic spells before traveling through the temple.

### CODE REVIEW

**Comment by drboldwin** [comment]
https://github.com/rpherbig/dr-scripts/issues/882

as well as accounting for cyclic spells before traveling through the temple.

File: theurgy.lic [hunk 1]
```diff
@@ -290,6 +290,10 @@ class Theurgy
 
   def study_wall
     return if DRStats.circle < 30
+    get_data('spells')
+      .spell_data
+      .select { |_name, properties| properties['cyclic'] }
+    return unless get_data('spells').spell_data.select { |name, _properties| DRSpells.active_spells.keys.include?(name) }.empty?
```
Comment (rpherbig) [inline theurgy.lic]: Is this logic used in any other scripts? I'm wondering if there would be use in moving it to common-arcana as a helper.

Comment (drboldwin) [inline theurgy.lic]: It's very similar to how common-arcana does release cyclics, but I'm only trying to check if a cyclic is running here, which doesn't have a method built into common-arcana.  Though we could push the method over there and just call it from here.  I wasn't sure how often we would need to only check if a cyclic was running.

Comment (rpherbig) [inline theurgy.lic]: If this is the only occurrence, I wouldn't bother making a helper yet.

Comment (drboldwin) [inline theurgy.lic]: Either way - it becomes fairly easy to do if we ever feel like it.

**Comment by drboldwin** [comment]
Alright well assuming we are keeping this by itself for now - I think the changes made to common-items will account for tapping someone - but this wont stop it from tapping the person, just register it as a fail.  If you are cool with that we can push this.

**Comment by rpherbig** [comment]
I'll have to ponder a better way to handle 'In' and 'My' people in the room. But in the meantime, you're right - this is better than what we had before.

**Pull request merged**